src/lib/IRTransponder/ILapTransponder.cpp

```diff
@@ -0,0 +1,144 @@
+// or using extrapolated values:
+// 26.1 * 60 = 1566
+// 1000000 / 1566 = 638.5696040868455 = 38314.176
+// assuming measurement error, clock drift or signal error caused by rounding on iLap transponder (which has a 7.372Mhz crystal)
+// 38400 * 12 = 460800 (carrier)
+//
+// Note: an ESP32 with an 80Mhz APB driving the RMT and a carrier frequency of 460.800kHz requires a clock divider of 173.11 (not a round number)
+// so 80Mhz / 173 = of 462.427kHz, not 460.800kHz, but an actual iLap receiver is able to receiver the signal.
+// and 462427 / 12 = 38535.583bps, not 38400bps.
+// thus, the code could be improved to calculate the bitrate based on the ACTUAL resolution of the RMT.
+// actual RMT resolution should be used to set the carrier_hz and the bitrate
+
+
+bool ILapTransponder::init()
+{
+    if (!transponderRMT->init(BITRATE * BIT_PERIODS, CARRIER_HZ, CARRIER_DUTY)) {
+      return false;
+    };
+
+    //uint64_t data = 0xFF0FEDCBA987; // NOT A VALID TRANSPONDER CODE
+    uint64_t data = config.GetIRiLapCode();
+
+    DBGLN("ILapTransponder::init, data: 0x%x", data);
+
+    encoder->encode(transponderRMT, data);
+
+    return true;
+}
+
+void ILapTransponder::deinit()
+{
+    DBGLN("ILapTransponder::deinit");
+    transponderRMT->deinit();
+}
+
+void ILapTransponder::startTransmission()
+{
+    transponderRMT->start();
+}
+
+void ILapEncoder::encode(TransponderRMT *transponderRMT, uint64_t data) {
+    generateBitStream(data);
+    bits_encoded = 0;
+    transponderRMT->encode(this);
+}
+
+bool ILapEncoder::encode_bit(rmt_item32_t *rmtItem) {
+
+    uint8_t bit = (bitStream & ((uint64_t)1 << NBITS)) > 0 ? 1 : 0;
+    DBGVLN("ILapEncoder::encode_bit, index: %d, bit: %d", bits_encoded, bit);
+
+    if (bit)
+    {
+        rmtItem->duration0 = 11;
+        rmtItem->level0 = 1;
+        rmtItem->duration1 = 1;
+        rmtItem->level1 = 0;
+    }
+    else
+    {
+        rmtItem->duration0 = 6;
+        rmtItem->level0 = 0;
+        rmtItem->duration1 = 6;
+        rmtItem->level1 = 0;
+    }
+
+    bitStream <<= 1;
+
+    bits_encoded ++;
+
+    bool done = bits_encoded == NBITS;
+
+    return done;
+}
+
+void ILapEncoder::generateBitStream(uint64_t data) {
+  uint8_t byte;
+  uint64_t mask;
+
+  bitStream = 0;
+
+  for (uint8_t i = 0; i < BYTES; i++) {
+    // start bit
+    bitStream |= 1; bitStream <<= 1;
+
+    byte = ((uint8_t *)&data)[ BYTES - i - 1];
+    DBGLN("ILapEncoder::generateBitStream, byte: 0x%x", byte);
+
+    mask = 0x01; // start with LSB
+
+    // 8 data bits, LSB first
+    for (uint8_t n = 0; n < 8; n++) {
+      uint8_t bit = 0;
+      if (!(byte & mask))
+        bit |= 1;
+
+      DBGVLN("ILapEncoder::generateBitStream, byte_index: %d, bit_of_byte: %d, byte: %x, bit: %d", i, n, byte, bit);
+
+      bitStream |= bit;
+      bitStream <<= 1;
+      mask <<= 1;
+    }
+
+    // no parity bit
+
+    //stop bit
+    bitStream <<= 1;
+  }
+  bitStream >>= 1;
+}
```

Comment:
Curly should be on the next line. This is the style we seem to have converged on. Please check this in all the changed/added files.